### DISCUSSION TITLE
How to use client side scripts

### DISCUSSION THREAD

**Discussion by austenstone** [freshframework/fresh#467] [Q&A]
Disclaimer: I'm a total noob with SSR

It seems [islands ](https://fresh.deno.dev/docs/getting-started/adding-interactivity) are the only place where I can ship JavaScript to the client.

I was attempting to import [FirebaseUI](https://firebase.google.com/docs/auth/web/firebaseui) in an island.
Something like 
```typescript
// islands/FirebaseUi.tsx
import * as firebaseui from "https://www.gstatic.com/firebasejs/ui/6.0.1/firebase-ui-auth.js";
```
But I am getting `error: ReferenceError: document is not defined` on the server.
I don't want to load this script on the server.

How can I import client side libraries such as firebaseui?

**Comment by PhoneDroid** [comment]
You can ensure code only runs on 
the `client` / `server` by using the 
`IS_BROWSER` variable.

```ts
import { IS_BROWSER } from '$fresh/runtime.ts'

if( IS_BROWSER )
    console.debug(`Client Side`)
else
    console.debug(`Server Side`)
```
